flink-metrics/flink-metrics-prometheus/src/test/java/org/apache/flink/metrics/prometheus/PrometheusReporterTest.java

```diff
@@ -85,7 +85,7 @@
     public void setupReporter() {
         registry =
                 new MetricRegistryImpl(
-                        MetricRegistryConfiguration.defaultMetricRegistryConfiguration(),
+                        MetricRegistryTestUtils.defaultMetricRegistryConfiguration(),
```

Comment:
This is a bit unrelated to this change. But it looks to me that this test is more of an integration test which could go into `flink-tests` because we use things from `flink-runtime`.

Reply:
Alternatively, this might indicate that the reporters are not super easy to test since we require the registry.

Reply:
The main issue I think is that we don't have a good MetricGroup for testing, so we have to rely on the runtime implementations which unfortunately rely on the registry. Overall the metric components are too reliant on each other.

I should have one such implementation somewhere, which should allow us to simplify quite a lot of metric-related tests, but I would do that as a follow-up because it implies bigger changes to all reporter tests.